zooniverse/front-end-monorepo
# lib-user: Add usePanoptesAuthUser, refactor related hooks and components

## Package
- lib-user

## Describe your changes
- add `usePanoptesAuthUser` hook for authorized user
- refactor `usePanoptesUser` hook to return user with `avatar_src` if don't already have user with `avatar_src` (almost adds functionality for an admin to see a different user's stats, that's done in #5980 )
- refactor hooks that make a request that uses authorization
- refactor to `useStats` hook per the source of stats expected (`sourceId`), generalizes hook for user or user group

## Context
Previously, lib-user hooks were inconsistently using the authClient passed as prop or the panoptes-javascript-client auth specifically. This created issues when running the dev server for local development and otherwise. These PR refactors make lib-user truly agnostic about the authClient (auth, oauth, other) passed to the components. 

Some related additional potential refactors could be to add React context for the authClient and authorized user (`authUser`), and/or to refactor lib-react-components `usePanoptesUser` to be authClient agnostic, though I don't think either of those are warranted at the moment. Alternatively, there's a fair amount of added complexity for lib-user to be authClient agnostic, maybe that should be revisited after launch and app-root is more established, in conjunction with larger FEM apps/library organization - not a problem for the short term, but noting for consideration in the long term.

## How to Review
_Helpful explanations that will make your reviewer happy:_
- What Zooniverse project should my reviewer use to review UX? n/a
- What user actions should my reviewer step through to review this PR?
  - review https://local.zooniverse.org:8080/?users=markb-panoptes/groups (replace markb-panoptes with your staging user login)
  - review https://local.zooniverse.org:8080/?users=markb-panoptes/stats (replace markb-panoptes with your staging user login)
  - review https://local.zooniverse.org:8080/?groups=1326945 (replace 1326945 with a user group you belong to)
- Which storybook stories should be reviewed? n/a
- Are there plans for follow up PR’s to further fix this bug or develop this feature? yes, see linked GitHub project, but specifically:
  - pass login in url param from app-root to lib-user components 
  - add containers to main components for cleaner code, can create stories and tests for components 
  - pass adminMode boolean from app-root to lib-user, update usePanoptesUser and usePanoptesUserGroup hooks accordingly
  - add sad paths (no authUser, no stats source, errors, loading, etc.) to containers
  - refactor GroupCards as links (with tab focus)

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General

## Refactoring
